src/wayland/dmabuf/mod.rs

```diff
@@ -288,15 +290,17 @@ pub enum ImportError {
     InvalidFormat,
 }
 
-/// Gets the contents of a [`Dmabuf`] backed [`Buffer`].
+/// Gets the contents of a [`Dmabuf`] backed [`WlBuffer`].
 ///
 /// If the buffer is managed by the dmabuf handler, the [`Dmabuf`] is returned.
 ///
 /// If the buffer is not managed by the dmabuf handler (whether the buffer is a different kind of buffer,
 /// such as an shm buffer or is not managed by smithay), this function will return an [`UnmanagedResource`]
 /// error.
-pub fn get_dmabuf(buffer: &Buffer) -> Result<Dmabuf, UnmanagedResource> {
-    Ok(buffer.buffer_data::<Dmabuf>()?.ok_or(UnmanagedResource)?.clone())
+///
+/// [`WlBuffer`]: wl_buffer::WlBuffer
+pub fn get_dmabuf(buffer: &wl_buffer::WlBuffer) -> Result<Dmabuf, UnmanagedResource> {
+    Ok(buffer.data::<Dmabuf>().ok_or(UnmanagedResource)?.clone())
```

Comment:
```suggestion
    buffer.data::<Dmabuf>().cloned().ok_or(UnmanagedResource)
```

Reply:
My rationale here is to avoid the clone if the data is not a dmabuf.

Reply:
Calling Cloned on None will just return None:
```rs
        match self {
            Some(t) => Some(t.clone()),
            None => None,
        }
```
So this is basically free, or is this just stylistic choice?

Reply:
Ah okay, so the clone is effectively free then.